code/game/objects/effects/spawners/bombspawner.dm

```diff
@@ -2,96 +2,57 @@
 	name = "bomb"
 	icon = 'icons/mob/screen_gen.dmi'
 	icon_state = "x"
-	var/btype = 0 // 0=radio, 1=prox, 2=time
 	var/btemp1 = 1500
 	var/btemp2 = 1000	// tank temperatures
 
-/obj/effect/spawner/newbomb/timer
-	btype = 2
-
-	syndicate
-		btemp1 = 150
-		btemp2 = 20
-
-/obj/effect/spawner/newbomb/proximity
-	btype = 1
-
-/obj/effect/spawner/newbomb/radio
-	btype = 0
-
-
-/obj/effect/spawner/newbomb/New()
+/obj/effect/spawner/newbomb/Initialize()
 	..()
+	var/obj/item/device/transfer_valve/V = new(src.loc)
+	var/obj/item/weapon/tank/internals/plasma/full/PT = new(V)
+	var/obj/item/weapon/tank/internals/oxygen/OT = new(V)
 
-	switch (src.btype)
-		// radio
-		if (0)
-
-			var/obj/item/device/transfer_valve/V = new(src.loc)
-			var/obj/item/weapon/tank/internals/plasma/PT = new(V)
-			var/obj/item/weapon/tank/internals/oxygen/OT = new(V)
-
-			var/obj/item/device/assembly/signaler/S = new(V)
-
-			V.tank_one = PT
-			V.tank_two = OT
-			V.attached_device = S
+	PT.air_contents.temperature = btemp1 + T0C
+	OT.air_contents.temperature = btemp2 + T0C
 
-			S.holder = V
-			S.toggle_secure()
-			PT.master = V
-			OT.master = V
+	V.tank_one = PT
+	V.tank_two = OT
+	PT.master = V
+	OT.master = V
+	
+	setup_assembly(V)
 
-			PT.air_contents.temperature = btemp1 + T0C
-			OT.air_contents.temperature = btemp2 + T0C
+	V.update_icon()
+	
+	qdel(src)
 
-			V.update_icon()
-
-		// proximity
-		if (1)
-
-			var/obj/item/device/transfer_valve/V = new(src.loc)
-			var/obj/item/weapon/tank/internals/plasma/PT = new(V)
-			var/obj/item/weapon/tank/internals/oxygen/OT = new(V)
-
-			var/obj/item/device/assembly/prox_sensor/P = new(V)
-
-			V.tank_one = PT
-			V.tank_two = OT
-			V.attached_device = P
-
-			P.holder = V
-			P.toggle_secure()
-			PT.master = V
-			OT.master = V
-
-
-			PT.air_contents.temperature = btemp1 + T0C
-			OT.air_contents.temperature = btemp2 + T0C
-
-			V.update_icon()
+/obj/effect/spawner/newbomb/proc/setup_assembly(obj/item/device/transfer_valve/V)
+	return
 
+/obj/effect/spawner/newbomb/timer
 
-		// timer
-		if (2)
-			var/obj/item/device/transfer_valve/V = new(src.loc)
-			var/obj/item/weapon/tank/internals/plasma/PT = new(V)
-			var/obj/item/weapon/tank/internals/oxygen/OT = new(V)
+/obj/effect/spawner/newbomb/timer/setup_assembly(obj/item/device/transfer_valve/V)
+	var/obj/item/device/assembly/timer/T = new(V)

```

Comment:
I suppose, wanted to leave it extendable but let's be honest no one is going to touch this code in next few years.